### DISCUSSION TITLE
入力モードが不意に直接入力になる。

### DISCUSSION THREAD

**Discussion by hayashida-katsutoshi** [google/mozc#907] [Q&A]
## Description
日本語(mozc)の入力モードがランダムに「ひらがな」から「直接入力」になる

## Steps to reproduce
1. 日本語と英語を、Super+SPACEで切り替えながら入力する。
2. しばらく繰り返していると「mozc」の入力モードがランダムに「直接入力」になる。

## Expected behavior
日本語(mozc)の入力モードは常に「ひらがな」で固定されて欲しい。

## Actual behavior
「mozc」の入力モードがランダムに「直接入力」になる。

## Screenshots
n/a

## Version or commit-id
2.26.4220.102

## Environment
 - Ubuntu 22.04.4 LTS
 - IBus
 - FeatherPad 1.0.1, Microsoft Edge

## Investigations
 - 英語キーボードを使用。変換、無変換キーはありません。

**Comment by yukawa** [comment]
`2.26.4220.102` is a quite old build, which we published more than 3 years ago. https://github.com/google/mozc/commit/9ba59b64d53365c1fe93c1c245b4ec3e35bdadf0

Until this is confirmed to be an issue that is still reproducible with the latest version of Mozc and is something that can/should be fixed in the Mozc side, let's talk about this in the Discussions pages as we would like to use Issues pages for something actionable.

**Comment by yukawa** [comment]
As far as I know, such an issue has never been reported in other platforms such as macOS and Windows, which makes me suspect the issue you observed was one of the following cases.
* It is an issue in [ibus](https://github.com/ibus/ibus) runtime.
* It is an issue in Gnome (and its extensions)
* It is an issue in `ibus-mozc`

Another thing we should be aware of is that even if the root cause was identified and fixed, you wont' receive the fix automatically as Ubuntu is not a rolling update OS. Here are necessary steps for you to be able to receive the **fix**.

1. Someone decides to take their time to work on this issue.
2. Succeed to reproduce the issue locally, which is most of the cases critical to identify the root cause.
3. Actually identify the root cause.
4. Come up with an idea on how to fix it in which component. The component to be fixed is not necessarily Mozc. It can be any other relevant component such as [ibus](https://github.com/ibus/ibus)
5. Create a proposed fix and get it reviewed by the component owner.
6. Get OK from the component owner to submit the fix.
7. Reach out to the package maintainer in Debian so that a fixed version of component can be imported to Debian unstable repository.
8. Wait until the fixed version of component is actually imported to the Debian unstable repository. This requires an approval from the Debian package maintainer.
9. Wait until the fixed version of component is imported to Ubuntu. If you only use Ubuntu LTS, the earliest version of Ubuntu that could contain the fix is most likely Ubuntu 26.04.
10. Finally you can update your Ubuntu OS to get the issue fixed.

We are currently waiting for the step 1 to happen.

**Comment by hayashida-katsutoshi** [comment]
I'd like to share some tips I've discovered. While this isn't a complete solution, it has made my life a bit easier.

Firstly, although you can switch between ENG and Mozc using Super+Space, I've decided not to do that. Instead, I keep the IME always set to Mozc and use CTRL+Space to toggle direct input on and off, to switch English and Japanese.

References:
https://qiita.com/kojix2/items/215c1477f38c6d2d7afa
https://magidropack.hatenablog.com/entry/2019/01/05/174048

Even when the IME is locked to Mozc, it still sometimes randomly switches to direct input mode. I believe this happens when switching between applications. In Mozc, the taskbar shows "A" for direct input mode and "あ" for Hiragana input mode. When the problem occurs, the taskbar still shows "あ," but it's actually acting in direct input mode. With the key binding change, you can press CTRL+Space to switch back to Hiragana input.
